test/integ_tests/test_device_creation.py

```diff
@@ -74,43 +75,61 @@ def test_get_devices_all():
         assert arn in result_arns
 
 
+def _get_provider_name(device: AwsDevice) -> str:
+    arn_provider = device.arn.split("/")[2]
+
+    # capitalize as in provider name
+    provider_primary_name = device.provider_name.split()[0]
```

Comment:
I'm not crazy about this heuristic, why can't we always just capitalize?

Reply:
Python uppercase variables are generally global constants, so I'm hesitant to use them for classes and/or constants that could one day become classes. For me, the tiebreaker that seals my vote is that changing the enum would be a breaking change and I don't feel strongly enough that this warrants that

Reply:
We could make an argument for the device names themselves being uppercase as enum values ([docs](https://docs.python.org/3/library/enum.html)), but that wouldn't solve this issue in specific